### PR TITLE
Razor SDK build ordering issues

### PR DESCRIPTION
**Pull request by pranavkm**
* Build the SDK completely regardless of the MSBuild runtime type
* Split SDK integration tests into a separate project. Only a handful of unit tests, test the SDK and they can continue referencing it. Integration tests can build the SDK project using `SkipGetTargetFrameworkProperties` to ensure the SDK is completely built before tests are run.

### CODE REVIEW

File: src/Components/Web.JS/dist/Release/blazor.server.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
...1),o=Object.keys||function(e){var t=[];for(var n in e)t.push(n);return t};e.exports=f;var i=n(19);i.inherits=n(14);var a=n(35),s=n(40);i.inherits(f,a);for(var c=o(s.prototype),u=0;u<c.length;u++){var l=c[u];f.prototype[l]||(f.prototype[l]=s.prototype[l])}function f(e){if(!(this instanceof f))return new f(e);a.call(this,e),s.call(this,e),e&&!1===e.readable&&(this.readable=!1),e&&!1===e.writable&&(this.writable=!1),this.allowHalfOpen=!0,e&&!1===e.allowHalfOpen&&(this.allowHalfOpen=!1),this.once("end",h)}function h(){this.allowHalfOpen||this._writableState.ended||r.nextTick(p,this)}function p(e){e.end()}Object.defineProperty(f.prototype,"writableHighWaterMark",{enumerable:!1,get:function(){return this._writableState.highWaterMark}}),Object.defineProperty(f.prototype,"destroyed",{get:function(){return void 0!==this._readableState&&void 0!==this._writableState&&(this._readableState.destroyed&&this._writableState.destroyed)},set:function(e){void 0!==this._readableState&&void 0!==this._writableState&&(this._readableState.destroyed=e,this._writableState.destroyed=e)}}),f.prototype._destroy=function(e,t){this.push(null),this.end(),r.nextTick(t,e)}},function(e,t,n){"use strict";(function(e){
```
Comment (JunTaoLuo) [inline src/Components/Web.JS/dist/Release/blazor.server.js:1]: Revert?

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj [hunk 2]
```diff
@@ -94,13 +95,11 @@
...
 
-    <Message Importance="High" Text="Prepared Razor SDK at $(SdkOutputPath)" />
-
-    <ItemGroup>
+    <ItemGroup Condition="'$(ContinuousIntegrationBuild)' == 'true'">
       <_PreparedRazorSdkFiles Include="$(SdkOutputPath)**\*.*" />
     </ItemGroup>
-
-    <Message Importance="High" Text="Prepared files:@(_PreparedRazorSdkFiles->'%0a%(Identity)')" />
+    <Message Importance="High" Text="Prepared Razor SDK at $(SdkOutputPath)" Condition="'$(ContinuousIntegrationBuild)' == 'true'" />
```
Comment (JunTaoLuo) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:101]: 👍

**Comment by pranavkm** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=682057,
https://dev.azure.com/dnceng/internal/_build/results?buildId=681329&view=results
https://dev.azure.com/dnceng/internal/_build/results?buildId=682058&view=results

**Comment by dougbu** [comment]
Am I right https://dev.azure.com/dnceng/internal/_build/results?buildId=682057&view=logs&j=5ac7b393-e840-5549-7fb4-a4479af8e7e3&t=29df2fa2-0d20-51bd-e85a-8b546e86c529&l=1298 is a variant of the same ordering issue❔
``` text
/datadisks/disk1/workspace/_work/1/s/src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets(54,5): error MSB4062: The "Microsoft.AspNetCore.Razor.Tasks.FindAssembliesWithReferencesTo" task could not be loaded from the assembly /datadisks/disk1/workspace/_work/1/s/artifacts/bin/Microsoft.NET.Sdk.Razor/Release/sdk-output/tasks/net5.0/Microsoft.NET.Sdk.Razor.Tasks.dll. Could not load file or assembly '/datadisks/disk1/workspace/_work/1/s/artifacts/bin/Microsoft.NET.Sdk.Razor/Release/sdk-output/tasks/net5.0/Microsoft.NET.Sdk.Razor.Tasks.dll'. The system cannot find the path specified. [/datadisks/disk1/workspace/_work/1/s/src/Security/Authentication/Cookies/samples/CookieSessionSample/CookieSessionSample.csproj]
/datadisks/disk1/workspace/_work/1/s/src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets(54,5): error MSB4062:  Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [/datadisks/disk1/workspace/_work/1/s/src/Security/Authentication/Cookies/samples/CookieSessionSample/CookieSessionSample.csproj]
```

**Comment by pranavkm** [comment]
It was. I tracked down what was causing the issue - missed clearing out `TargetFrameworks` (plural) in CSharp.Common.props. In additiona, I was also able to change to use vanilla project references and remove explicit calls to run build and publish. Let's see how that works.

**Comment by wtgodbe** [comment]
New buildz:

https://dev.azure.com/dnceng/internal/_build/results?buildId=682898&view=results
https://dev.azure.com/dnceng/internal/_build/results?buildId=682899&view=results
https://dev.azure.com/dnceng/internal/_build/results?buildId=682900&view=results

File: src/Components/Web.JS/dist/Release/blazor.server.js [hunk 3]
```diff
@@ -1,19 +1,19 @@
...1),o=Object.keys||function(e){var t=[];for(var n in e)t.push(n);return t};e.exports=f;var i=n(19);i.inherits=n(15);var a=n(35),s=n(40);i.inherits(f,a);for(var c=o(s.prototype),u=0;u<c.length;u++){var l=c[u];f.prototype[l]||(f.prototype[l]=s.prototype[l])}function f(e){if(!(this instanceof f))return new f(e);a.call(this,e),s.call(this,e),e&&!1===e.readable&&(this.readable=!1),e&&!1===e.writable&&(this.writable=!1),this.allowHalfOpen=!0,e&&!1===e.allowHalfOpen&&(this.allowHalfOpen=!1),this.once("end",h)}function h(){this.allowHalfOpen||this._writableState.ended||r.nextTick(p,this)}function p(e){e.end()}Object.defineProperty(f.prototype,"writableHighWaterMark",{enumerable:!1,get:function(){return this._writableState.highWaterMark}}),Object.defineProperty(f.prototype,"destroyed",{get:function(){return void 0!==this._readableState&&void 0!==this._writableState&&(this._readableState.destroyed&&this._writableState.destroyed)},set:function(e){void 0!==this._readableState&&void 0!==this._writableState&&(this._readableState.destroyed=e,this._writableState.destroyed=e)}}),f.prototype._destroy=function(e,t){this.push(null),this.end(),r.nextTick(t,e)}},function(e,t,n){"use strict";(function(e){
```
Comment (JunTaoLuo) [inline src/Components/Web.JS/dist/Release/blazor.server.js:1]: revert again

Comment (pranavkm) [reply src/Components/Web.JS/dist/Release/blazor.server.js:1]: Just ignore it. The file in source control is incorrect which is the only reason this changes every time we rebuild.

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj [hunk 4]
```diff
@@ -26,8 +26,22 @@
...
+      Include="..\..\Microsoft.AspNetCore.Razor.Tools\src\Microsoft.AspNetCore.Razor.Tools.csproj"
+      Targets="Publish"
+      ReferenceOutputAssembly="false"
+      IsImplicityDefined="false"
+      SkipGetTargetFrameworkProperties="true"
+      UndefineProperties="TargetFramework;TargetFrameworks" />
+
+    <ProjectReference
+      Include="..\..\Microsoft.AspNetCore.Mvc.Razor.Extensions\src\Microsoft.AspNetCore.Mvc.Razor.Extensions.csproj"
+      Targets="Publish"
```
Comment (wtgodbe) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:39]: Didn't know you could do this... cool

**Review by wtgodbe — APPROVED** [review]

File: src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj [hunk 5]
```diff
@@ -26,8 +26,22 @@
...
+      ReferenceOutputAssembly="false"
+      IsImplicityDefined="false"
+      SkipGetTargetFrameworkProperties="true"
+      UndefineProperties="TargetFramework;TargetFrameworks" />
```
Comment (dougbu) [inline src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:43]: Not a big deal but this should work using `@(Reference)` items too

Comment (pranavkm) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:43]: In the WASM branch, I had trouble getting `IncludeAssets="compile"` to work with reference, so I assumed it only supports a subset of things ProjectReferences do

Comment (dougbu) [reply src/Razor/Microsoft.NET.Sdk.Razor/src/Microsoft.NET.Sdk.Razor.csproj:43]: That's not an issue in 'master' because we use the JoinItems task.

**Pull request merged**